ui_framework/components/button/button.test.js

```diff
@@ -34,11 +34,11 @@ describe('KuiButton', () => {
   });
 
   describe('Props', () => {
-    describe('type', () => {
-      BUTTON_TYPES.forEach(type => {
-        describe(type, () => {
-          test(`renders the ${type} class`, () => {
-            const $button = render(<KuiButton type={type} />);
+    describe('buttonType', () => {
+      BUTTON_TYPES.forEach(buttonType => {
+        describe(buttonType, () => {
+          test(`renders the ${buttonType} class`, () => {
+            const $button = render(<KuiButton buttonType={buttonType} />);
```

Comment:
We'll need to update the test above this one to be:

```javascript
    test('HTML attributes are rendered', () => {
      const $button = render(
        <KuiButton
          aria-label="aria label"
          className="testClass1 testClass2"
          data-test-subj="test subject string"
          type="submit"
          disabled
        />
      );

      expect($button)
        .toMatchSnapshot();
    });
```

This will allow us to test that specifying the prop `type="submit"` renders as the HTML attribute `type="submit"`.

We'll need to do this with the KuiLinkButton and KuiSubmitButton tests. too.